### PR TITLE
Add Libwebsockets package

### PR DESCRIPTION
**Pull request by chetant**

### CODE REVIEW

File: pkgs/development/libraries/libwebsockets/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ fetchgit, stdenv, cmake, openssl, zlib }:
+
+stdenv.mkDerivation rec {
+  name = "libwebsockets-1.4";
+
+  src = fetchgit {
```
Comment (pSub) [inline pkgs/development/libraries/libwebsockets/default.nix]: Are there no release tar-balls available?

File: pkgs/development/libraries/libwebsockets/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    description = "Light, portable C library for websockets";
+    longDescription = ''
+      Libwebsockets is a lightweight pure C library built to
+      use minimal CPU and memory resources, and provide fast
+      throughput in both directions.
+    '';
+    homepage = https://libwebsockets.org/trac/libwebsockets;
+    # See http://git.libwebsockets.org/cgi-bin/cgit/libwebsockets/tree/LICENSE
+    license = stdenv.lib.licenses.gpl2Plus;
+    maintainers = [ ];
```
Comment (pSub) [inline pkgs/development/libraries/libwebsockets/default.nix]: Do you mind to add yourself as maintainer?

File: pkgs/development/libraries/libwebsockets/default.nix (hunk 1 above)
Comment (chetant) [inline pkgs/development/libraries/libwebsockets/default.nix]: I couldn't find any on the official site.

File: pkgs/development/libraries/libwebsockets/default.nix (hunk 2 above)
Comment (chetant) [inline pkgs/development/libraries/libwebsockets/default.nix]: Given my time constraints, I'm not sure I'll be able to maintain this package.

**Comment by lethalman** [comment]
Built locally thanks.

**Pull request merged**